File: leetcode/31. maxProduct/maxProduct.go
package maxproduct

func maxProduct(nums []int) int {
	a, b := 0, 0
	for _, v := range nums {
		if v > a {
			a, b = v, a
		} else if v > b {
			b = v
		}
	}
	return (a - 1) * (b - 1)
}

File: leetcode/31. maxProduct/maxProduct.py
from typing import List

def maxProduct(nums: List[int]) -> int:
    a, b = 0, 0
    for num in nums:
        if num > a:
            a, b = num, a
        elif num > b:
            b = num
    return (a - 1) * (b - 1)